src/main/scala/com/atomist/rug/BadRugException.scala

```diff
@@ -82,6 +84,24 @@ class BadPlanException(msg: String)
 class MissingRugException(msg: String)
   extends BadRugException(msg)
 
+class EditorNotFoundException(msg: String)
+  extends BadRugException(msg) {
+  def this(requestedEditorName: String, knownRugs: Seq[Rug]) = {
+    this(EditorNotFoundExceptionStr(requestedEditorName, knownRugs))
+  }
+}
+
+
+/**
+  * Generate a an exception string containing all known editors.
+  */
+private object EditorNotFoundExceptionStr {
+  def apply(requestedEditorName: String, knownRugs: Seq[Rug]): String = {
+    s"""|Could not find editor [$requestedEditorName]. Known editors:
+        |[${knownRugs.collect{case o: ProjectEditor => o}.map(p => p.name).mkString(",")}]""".stripMargin
+  }
+}
+
```

Comment:
I wonder how this will look and I'm not sure this will be that helpful with large catalogs of locally installed editors. But let's worry about it when that's the case I guess :)